platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileReader.java

```diff
@@ -103,7 +104,7 @@ public static List<SavedStateInfo> getSavedStateFiles(
         final SignedState newSignedState = new SignedState(
                 configuration,
                 CryptoStatic::verifySignature,
-                normalizedData.state(),
+                (MerkleRoot) normalizedData.state(),
```

Comment:
why is this being cast? does the signed state need to change?

Reply:
This is a temporary fix. We have a ticket for a final solution about `MerkleRoot` interface - https://github.com/hashgraph/hedera-services/issues/16436. In this case we need `MerkleRoot` instance and `PlatformMerkleStateRoot` _is_ an instance of `MerkleRoot`, so I don't anticipate any issues here.